### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by dgkncelik**

### CODE REVIEW

File: src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+    @Autowired
+    private IcalService icalService;
+
+    @Autowired
+    private ParticipantService participantService;
+
+    @Override
+    public void addParticipant(Participant participant){
+         if(participant.getUserName().isEmpty() || participant.getUserName() == null){
+             throw new ParticipantConflict(1000,"Participant username cannot be empty");
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java:30]: Do we really need error code? If yes, tracking the error codes hard in this case. Maybe we can consider removing.

File: src/main/java/edu/itu/cavabunga/controller/ParticipantController.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+public class ParticipantController {
+   @Autowired
+   private CalendarManagerService calendarManagerService;
+
+   @Autowired
+   private ParticipantResponse participantResponse;
+
+   @GetMapping(path = "/testcalendarcreate")
+   @ResponseStatus(HttpStatus.OK)
+   public @ResponseBody String test(){
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/ParticipantController.java]: delete before merge

File: src/main/java/edu/itu/cavabunga/controller/wrapper/AuthenticationReponse.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package edu.itu.cavabunga.controller.wrapper;
+
+import edu.itu.cavabunga.core.entity.Authentication;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthenticationReponse {
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/wrapper/AuthenticationReponse.java:8]: I prefer all the auth component as another PR.

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ParameterResponse {
+    private Integer status;
+    private String message;
+    private List<Parameter> data = new ArrayList<Parameter>();
+
+    public Integer getStatus() {
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java:15]: do we really need the status fields in these response class? Why is HTTP status codes not enough?

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    public String message;
+    public List<ComponentProperty> data = new ArrayList<ComponentProperty>();
+
+    public ComponentPropertyResponse(Integer status, String message, List<ComponentProperty> data) {
+        this.status = status;
+        this.message = message;
+        this.data = data;
+    }
+
+    public Integer getStatus() {
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java:19]: Why is HTPP status codes not enough?

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
+package edu.itu.cavabunga.controller.wrapper;
+
+import edu.itu.cavabunga.core.entity.ComponentProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ComponentPropertyResponse {
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java:8]: All the response class almost same, they should inherit from a base response class

File: src/main/java/edu/itu/cavabunga/core/entity/Authentication.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+import edu.itu.cavabunga.core.entity.authentication.AuthenticationType;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Authentication {
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/core/entity/Authentication.java:11]: Authentication classes are not related to this pr.

File: src/test/java/edu/itu/cavabunga/core/factory/PropertyFactoryImplTest.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                {PropertyType.Status,Status.class},
+                {PropertyType.Summary,Summary.class},
+                {PropertyType.Transp,Transp.class},
+                {PropertyType.Trigger,Trigger.class},
+                {PropertyType.Tzid,Tzid.class},
+                {PropertyType.Tzname,Tzname.class},
+                {PropertyType.Tzoffsetfrom,Tzoffsetfrom.class},
+                {PropertyType.Tzoffsetto,Tzoffsetto.class},
+                {PropertyType.Tzurl,Tzurl.class},
+                {PropertyType.Uid,Uid	.class	},
```
Comment (karakayasemi) [inline src/test/java/edu/itu/cavabunga/core/factory/PropertyFactoryImplTest.java]: typo

**Review by karakayasemi — CHANGES_REQUESTED** [review]
the first part of my review.

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1,3 +1,3 @@
-spring.jpa.hibernate.ddl-auto=create
-spring.datasource.url=jdbc:mysql://127.0.0.1/test
+spring.jpa.hibernate.ddl-auto=update
```
Comment (karakayasemi) [inline src/main/resources/application.properties]: travis does not work with these application properties.

File: pom.xml [hunk 10]
```diff
@@ -59,7 +59,12 @@
 			<version>2.0</version>
 			<scope>test</scope>
 		</dependency>
-
+		<dependency>
```
[the remark this answers is not in the document]
Comment (dgkncelik) [reply pom.xml]: Removed with #bafe328 commit

File: src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java (hunk 1 above)
Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java:30]: For instance, in many place "checking participant exists" operation happens. Do determine where exactly exception comes from is important. <imho>

File: src/main/java/edu/itu/cavabunga/controller/ParticipantController.java (hunk 2 above)
Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/controller/ParticipantController.java]: Removed

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java (hunk 4 above)
Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java:15]: Status is changing with error codes. You may have the same 404 Http Code(which states for not found) for differ cases. Status code gives you idea that which error code you have in application to determine problem about request or operation.

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java (hunk 6 above)
Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java:8]: The problem about this thing is little different. Kind a bug is exists in Jackson library(people also complainig about it on github) to force me to do this. For example, if you wanna call a calendar object which has event sub object. Jackson will create a "@type: XXXX" column in the json response for all nested and parent object. Library also expects that column when converts http request json data to objects. Basically that "@type: XXX" thing is needed to determine which class you have. When you add object into response class and send as a http response. Parent object's "@type: XXX" json column dissapears(only the parent's @type column). So you should but objects in the type of your object's type array(or list or whatever)  to appear correctly in the http response. Its little bit tricky but i'll consider your review and will correct that

File: src/test/java/edu/itu/cavabunga/core/factory/PropertyFactoryImplTest.java (hunk 8 above)
Comment (dgkncelik) [reply src/test/java/edu/itu/cavabunga/core/factory/PropertyFactoryImplTest.java]: Fixed

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
+package edu.itu.cavabunga.controller.wrapper;
```
[the remark this answers is not in the document]
Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/controller/wrapper/ComponentPropertyResponse.java:1]: wrapping http responses as status, message, data  format

**Comment by dgkncelik** [comment]
Not merged.
Please try again later :)

File: src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java (hunk 1 above)
Comment (karakayasemi) [reply src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java:30]: You can use stack trace to determine where exactly exception comes from. Giving a random number does not help in my opinion.

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java (hunk 4 above)
Comment (karakayasemi) [reply src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java:15]: IMHO, you can use 404 for all missing component, parameter etc. Maybe I am missing something, we should redesign these response classes together.

File: src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+    public void addData(Parameter parameter){
+        this.data.add(parameter);
+    }
+
+    public void setDataNull(){
+        this.data = null;
+    }
+
+    public ParameterResponse createParameterResponseForList(Integer code, String message, List<Parameter> parameters){
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/wrapper/ParameterResponse.java:47]: function is same with createParameterReponseForSingle. Also, why an object creates an object from its own class?

**Review by karakayasemi — COMMENTED** [review]
I will review all of them in the #21. No need to use closed PR

**Pull request closed without merging**